app/controllers/general_infos_controller.rb

```diff
@@ -38,7 +36,13 @@ def update
   private
 
     def set_general_info
-      @general_info = @incident.general_info.target || GeneralInfo.new
+      if @incident.general_info.present?
+        @general_info = @incident.general_info.target
+      else
+        @general_info = GeneralInfo.new(incident: @incident)
+        # Setting general_info.incident causes validation to run, so clear those errors.
```

Comment:
Even when it's only set in a `new` constructor?? That seems pretty terrible, Dynamoid...